mod/storage/pkg/deposit/store.go

```diff
@@ -67,13 +67,15 @@ func (kv *KVStore[DepositT]) GetDepositsByIndex(
 	deposits := []DepositT{}
 	for i := range numView {
 		deposit, err := kv.store.Get(context.TODO(), startIndex+i)
-		if errors.Is(err, sdkcollections.ErrNotFound) {
+		switch {
+		case err == nil:
+			deposits = append(deposits, deposit)
+		case errors.Is(err, sdkcollections.ErrNotFound):
+			// not more deposits from index i on.
 			return deposits, nil
+		default:
+			return nil, err
```

Comment:
technically a change since upon err we don't return deposits cumulated so far. However err client does drop all deposits upon error, so we're fine